coffee_github/src/repository.rs

```diff
@@ -253,14 +253,13 @@ impl Repository for Github {
             }
         }
         // pull the changes from the repository
-        let status = fast_forward(&self.url.path_string, &self.branch)?;
-        // update the git information
-        // (git HEAD and date of the last commit)
-        let repo = git2::Repository::open(&self.url.path_string)
-            .map_err(|err| error!("{}", err.message()))?;
-        let (commit, date) = get_repo_info!(repo);
-        self.git_head = Some(commit.clone());
-        self.last_activity = Some(date.clone());
+        let status = git_upgrade(&self.url.path_string, &self.branch).await?;
+        // FIXME: add this code inside a method of the enum, to be able to call this as `status.date()`
+        let (_, date) = match status {
+            UpgradeStatus::UpToDate(ref commit, ref date) => (commit, date.clone()),
+            UpgradeStatus::Updated(ref commit, ref date) => (commit, date.clone()),
+        };
+        self.last_activity = Some(date);
```

Comment:
We should also update `self.git_head`

Reply:
uh why it was not updated before! good catch